### ISSUE TITLE
Collection interval adjustment doesn't take configured interval into account.

### ISSUE THREAD

**Issue by torbiak** [CryptonZylog/carbonator#3]
The collection interval adjustment in [CarbonatorInstance.cs:168](https://github.com/CryptonZylog/carbonator/blob/master/Carbonator/CarbonatorInstance.cs#L168) doesn't take ``configuration.carbonator.collectionInterval`` into account, instead subtracting the time spent collecting from 1000ms. This was missed when the collection interval was added to the config in 5a5fe6b7.

Also, I don't understand why the due time is reset to 100ms. I'm seeing continual interval adjustments, resulting in up to 10 collections per second. Shouldn't the due time be set to the new collection interval, instead?

Finally, I don't see why the interval adjustment is needed at all or how it would result in more accurate data as claimed in the comment at [CarbonatorInstance.cs:114](https://github.com/CryptonZylog/carbonator/blob/master/Carbonator/CarbonatorInstance.cs#L114). Please explain.

Thanks for your work on this. It's been a pleasure to use apart from the perplexing extra messages caused by the interval adjustment.

**Comment by nward** [comment]
Hi @torbiak,

This is exactly the fix I was looking for. I don't have Windows build tools handy, but noticed you've fixed this in your fork. I don't suppose you can upload a built version in lieu of a fixed release from @CryptonZylog?

**Comment by torbiak** [comment]
@nward: https://github.com/torbiak/carbonator/releases/tag/1.4-no_adj

**Comment by CryptonZylog** [comment]
Hi, been busy past couple of months here, sorry I could not respond earlier!

You're right, looks like this was missed during the update.

The self-adjustment was added after I noticed some latency from Carbonator when it was reading/processing many performance counters during each tick interval of ``collectMetrics``. I suppose this self-adjustment can be entirely removed. The collect interval can stay configurable for people that want to collect data every second, 5 seconds, etc.
